### DISCUSSION TITLE
Multiple partial indexes

### DISCUSSION THREAD

**Discussion by p53** [apache/couchdb#3284] [Q&A] [answered]
Hi, i have find query like this:

'{
  "selector": {
    "name": { "$gt": null },
    "_id": { "$gt": null },
    "owner": { "$gt": null },
    "org": { "$eq": "1111" },
    "state": { "$ne": "CarArchived" }
  },
  "limit": 10,
  "sort": [
    {"name": "asc"}
  ],
  "use_index": [
    "car_org_by_name",
    "car_org_purged_by_name"
  ]
}'

and getting error :

`{"error":"no_usable_index","reason":"No index exists for this sort, try indexing by the sort fields."}`

Here are my indexes:
car_org_by_name:

`{
			"fields": ["name", "org", "_id", "owner", "state"],
			"partial_filter_selector": {
				"state": {
					"$ne": "CarDeleted"
				}
			}
		}`

and 

car_org_purged_by_name:

`{
			"fields": ["name", "org", "_id", "owner", "state"],
			"partial_filter_selector": {
				"state": {
					"$ne": "CarPurged"
				}
			}
		}`

If i use just one index, any of those i don't get that error, problem arises when i want to use both of them

**Comment by willholley** [comment, marked as answer]
@p53 CouchDB only supports using one index per query. The `use_index` parameter only [accepts an array](https://docs.couchdb.org/en/stable/api/database/find.html#db-find) to allow the index name to be qualified by a design document name as well (in the case that you have multiple indices with the same name but defined in different design documents).

The error is not particularly intuitive, I agree. What's happening is that the query engine is looking for an index in design document `car_org_by_name` named `car_org_purged_by_name`, not finding one, and then trying to fall back to any valid index that it can find. Partial indexes are excluded from the automatic fallback, so you get an error that there are no valid indexes available for the query.

**Reply by p53** [reply]
ahh right, sorry overlooked that, thanks
